cmd/k8s-operator/proxygroup.go

```diff
@@ -153,6 +160,10 @@ func (r *ProxyGroupReconciler) reconcilePG(ctx context.Context, pg *tsapi.ProxyG
 		}
 	}
 
+	if err := r.validate(ctx, pg); err != nil {
```

Comment:
Invalid ProxyGroup should not result in the repeated reconciles (because of an error returned by the Reconcile func) - we should return nil from Reconcile and make sure that the reconciler is triggered again when the state is fixed (in this case when the ServiceAccount is created)